code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -178,6 +178,16 @@
 	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "chemical_laughter", /datum/mood_event/chemical_laughter)
 	..()
 
+/datum/reagent/consumable/laughter/expose_mob(mob/living/M, method=TOUCH, reac_volume)
+	var/mob/living/carbon/human/reactor = M
+	if(istype(reactor))
+		var/datum/component/mood/mood = reactor.GetComponent(/datum/component/mood)
+		if (mood.get_event("slipped"))
```

Comment:
dont use getcomponent to check specific events like this.

Reply:
What do you suggest?

Reply:
give return values on clearing events so you know if it actually cleared the event or not (aka it didnt exist), then you can just check what was sent back.

girls who dont put return values on their procs are h*tler

Reply:
that is just as awful

Reply:
make the slipped mood event listen for expose reagent on the mob and if so, make it give these effects.

Reply:
whats the benefit when you are a lot more likely to be slipped than to have laughter in you? You are more likely to be slipped without laughter than to have laughter in you but not be slipped (post PR definitely), so im confused if the reason is performance.

From OOP i would think you would want the interaction on the reagent vs. the slipping anyways.

Mind you this is from someone not with a coding background, im just confused on the WHY here.